### ISSUE TITLE
multiplerecordsfound

### ISSUE THREAD

**Issue by Bettzu** [m-schwabe/enrol_badgeenrol#2]
We have an additional role with archetype=student. Adding your enrolment method failes with the error message:

```
Debug info: SELECT id FROM {role} WHERE archetype = ?
[array (
0 => 'student',
)]
Error code: multiplerecordsfound
Stack trace:
    line 1490 of \lib\dml\moodle_database.php: dml_multiple_records_exception thrown
    line 1553 of \lib\dml\moodle_database.php: call to moodle_database->get_record_sql()
    line 1534 of \lib\dml\moodle_database.php: call to moodle_database->get_field_sql()
    line 1513 of \lib\dml\moodle_database.php: call to moodle_database->get_field_select()
    line 58 of \enrol\badgeenrol\edit_form.php: call to moodle_database->get_field()
    line 196 of \lib\formslib.php: call to enrol_badgeenrol_edit_form->definition()
    line 63 of \enrol\badgeenrol\edit.php: call to moodleform->moodleform()
```

**Comment by lebz** [comment]
Good day, did anyone find a solution to this? I am receiving the same error.

**Comment by ignacioduque** [comment]
Hi, i don't know if this is useful yet, but i found a workaround to this problem.
When the plugin tries to check for the `archetype = student`, if the system has more than one role created using the `student` archetype, it stores it in the database in the table `role` with the archetype `student`, so the result from the database call loads more than one record.
I checked the DB and find that the `shortname` column has a unique name, so i changed the DB check from `archetype = student` to `shortname = student` and now it works.

I found it in `enrol/badgeenrol/edit_form.php` on line 58.
It says `$studentid = $DB->get_field('role', 'id', array('archetype' => 'student'), MUST_EXIST);`
It should says `$studentid = $DB->get_field('role', 'id', array('shortname' => 'student'), MUST_EXIST);`

Hope it helps.
